cut/copy/paste

evgenykochetkov opened the pull request:
Closes #38

nkrkv commented:
Regarding the code :+1: 

I’ve tried to test the feature and all aspects seem to be solid and stable. Good job! Although I found few minor flaws that could be worked on now or later (in a dedicated tweak issue). It’s up to you to decide whether they could be tweaked in few hours or not.

1. If the clipboard has improper data (wouldn’t be pasted) than the paste operation should not deselect all. Try it a text editor.
2. If I copy then paste on the same patch I expect the pasted cluster to be at the same position as original with a little offset to the left and bottom (X ~ 1 pin width, Y ~ half of the node height), not viewport origin as currently
3. If I copy-paste some nodes, I expect the cluster to automatically include all links touching these nodes

brusherru commented on packages/xod-client/src/editor/utils.js:
```diff
@@ -27,3 +34,116 @@ export const getSelectedEntityIdsOfType = R.curry((entityType, selection) => R.c
+  })(entities);
+};
+
+// TODO: move to xod-func-tools?
```
Good idea :)

evgenykochetkov commented:
PR is ready 🕺

@nkrkv @brusherru check it out

- added Edge compatibility
- implemented "If I copy-paste some nodes, I expect the cluster to automatically include all links touching these nodes"

"If I copy then paste on the same patch I expect the pasted cluster to be at the same position as original with a little offset to the left and bottom (X ~ 1 pin width, Y ~ half of the node height), not viewport origin as currently" will be done as a separate tweak issue
